dask/dataframe/partitionquantiles.py

```diff
@@ -425,12 +431,25 @@ def percentiles_summary(df, num_old, num_new, upsample, state):
     elif is_datetime64_dtype(data.dtype) or is_integer_dtype(data.dtype):
         interpolation = "nearest"
 
-    # FIXME: pandas quantile doesn't work with some data types (e.g. strings).
-    # We fall back to an ndarray as a workaround.
+    # FIXME: Series.quantile doesn't work with some data types (e.g. strings).
+    # We try using DataFrame.quantile(..., method="table") as a workaround.
     try:
         vals = data.quantile(q=qs / 100, interpolation=interpolation).values
     except (TypeError, NotImplementedError):
-        vals, _ = _percentile(array_safe(data, like=data.values), qs, interpolation)
+        if PANDAS_GE_150 or not isinstance(data, pd.Series):
```

Comment:
@phofl - I'm interested in your thoughts on this. The original `_percentile` code path does not work for `cudf` string columns, because the `data.values` operation is not supported in `cupy`.

Do you expect the `*.quantile(..., method="table")` approach to affect performance for `pandas` in a non-negligible way? I'd like to avoid any new dispatching if this approach is suitable for both `pandas` and `cudf`. I was assuming that the shuffle will far outweigh any small change here, but I could be wrong.

Reply:
I think you'd have to run a quick benchmark, I am not really familiar how quantile is implemented internally.